poset/poset.go

```diff
@@ -186,17 +186,6 @@ func (p *Poset) processKnownRoots() *election.Res {
 // Event order matter: parents first.
 // ProcessEvent is not safe for concurrent use.
 func (p *Poset) ProcessEvent(e *inter.Event) (err error) {
-	was := p.EpochN
-
-	defer func() {
-		if err != nil {
-			return
-		}
-		immediately := (p.EpochN != was)
-		immediately = true // TODO: fix TestRestore then remove
-		p.store.Commit(e.Hash(), immediately)
-	}()
-
```

Comment:
check lines 208-212 in calcFrameIdx. If len(parents) == 0 AND not the first event, then error. Need to handle this error case.

Reply:
It's checked by basic_check that len(parents)>0 if e.Seq > 1. All the checkers were called during event downloading, before the poset is called

Reply:
- ok, it's checked by fetcher.
- btw, checkInited() may use constants instead of math.MaxInt32/2 and math.MaxInt64/2.